lib/ipaddr.rb

```diff
@@ -227,6 +227,22 @@ def to_string
     return str
   end
 
+  # Returns a string containing the IP address representation with prefix.
+  def as_json(*)
+    if ipv4? && prefix == 32
+      to_s
+    elsif ipv6? && prefix == 128
+      to_s
+    else
+      cidr
+    end
+  end
+
+  # Returns a json string containing the IP address representation.
+  def to_json(*args)
+    format("\"%s\"", as_json)
```

Comment:
Was your intent `as_json(*args)` because you gave it a name?

Reply:
@knu It is not necessary a name. I have removed a name of `args`.
It is from another patch which is https://github.com/ruby/json/pull/504/commits/f1ffc4a6187d983b89e33bd0520ad7ac3bba6fbf

Reply:
I prefer the implementation of to_json to be `as_json.to_json(*args)` so it is easier to change how IPAddrs are encoded simply by overriding `as_json`.  With the current definition, when you want it to be `def as_json(*) = to_f` you'll need to rewrite `to_json` as well.